LoopKit/GlucoseKit/GlucoseStore.swift

```diff
@@ -556,13 +556,29 @@ extension GlucoseStore {
     ///   - start: The earliest date of glucose values to include
     ///   - end: The latest date of glucose values to include, if provided
     ///   - effects: Glucose effects to be countered, in chronological order
+    ///   - unstoredGlucoseSample: A sample not included in the glucose store to be included in the computation of counteraction effects
     ///   - completion: A closure called once the values have been retrieved
     ///   - effects: An array of velocities describing the change in glucose samples compared to the specified effects
-    public func getCounteractionEffects(start: Date, end: Date? = nil, to effects: [GlucoseEffect], _ completion: @escaping (_ effects: [GlucoseEffectVelocity]) -> Void) {
+    public func getCounteractionEffects(
+        start: Date,
+        end: Date? = nil,
+        to effects: [GlucoseEffect],
+        unstoredGlucoseSample: NewGlucoseSample? = nil,
+        _ completion: @escaping (_ effects: [GlucoseEffectVelocity]) -> Void
+    ) {
         getCachedGlucoseSamples(start: start, end: end) { (samples) in
-            completion(samples.counteractionEffects(to: effects))
+            completion(self.counteractionEffects(for: samples, to: effects))
```

Comment:
Shouldn't unstoredGlucoseSample be added to samples here?

Reply:
Ah, shoot, thanks; I refactored based on Darin's comments (the logic for `unstoredGlucoseSample` has moved up to Loop) but forgot to remove the argument!